docs/zkapps/advanced-snarkyjs/actions-and-reducer.mdx

```diff
@@ -99,9 +99,63 @@ this.reducer.getActions({ fromActionsHash?: Field, endActionsHash?: Field }): A[
 
 Right now, `getActions` is available for [testing with `LocalBlockchain`](/zkapps/how-to-test-a-zkapp).
 
+### Fetching Actions from an Archive Node
+
+Since Mina nodes do not store historical network information and actions are not kept on-chain, emitted actions are only preserved in the Archive Node. Consequently, a zkApp must fetch its previously emitted actions on the Mina network from an Archive Node.
+
+As the Archive Node is essentially a database, it does not offer a user-friendly API that SnarkyJS can readily utilize to fetch actions. To address this issue, a GraphQL server was developed to retrieve network information such as actions and actions from the Archive Node. SnarkyJS will use this GraphQL server to fetch actions from the Archive Node.
+
+To configure SnarkyJS to use an Archive Node to fetch actions, use the `archive` property in the `Mina.Network` function. The `archive` property must be specified in the `Mina.Network` function to fetch actions from an Archive Node; otherwise, an error will occur.
+
+You can set the `archive` property to one of the following endpoints:
+
+```sh
+https://api.minascan.io/archive/berkeley/v1/graphql
+```
+
+```sh
+https://archive.berkeley.minaexplorer.com
+```
+
+For example, to use the `archive` property, see the following example code:
+
+```ts
+const Network = Mina.Network({
+  mina: '', // Use https://proxy.berkeley.minaexplorer.com/graphql or https://api.minascan.io/node/berkeley/v1/graphql
+  archive: '', // Use https://api.minascan.io/archive/berkeley/v1/graphql/ or https://archive.berkeley.minaexplorer.com/
+});
+Mina.setActiveInstance(Network);
+```
+
+By setting the `archive` property to one of the listed endpoints, SnarkyJS uses the selected endpoint to fetch events. For instance, if a zkApp has already emitted events and broadcasted the transaction, you can use the `getActions()` function to retrieve actions for a zkApp:
```

Comment:
Could we simply this to? 

> Within your smart contract, you can now use `getActions()` to retrieve actions emitted by your smart contract as part of previous transactions.

Reply:
@barriebyron Goal here is that they have to have done the preceeding step in order for this to be possible. That's the goal of now. Maybe you can suggest an alternative?